File: Problems/hacker_rank/Search/20150122_Ice_Cream_Parlor/Solution.scala
object Solution {
  def main(args: Array[String]) {
    val inps = io.Source.stdin.getLines.toArray
    val t = inps(0).toInt
    for ( i <- 1 to t * 3 by 3 )  {
      val m = inps(i).toInt
      val n = inps(i + 1).toInt
      val flavors = inps(i + 2).split(" ").map((s: String) => s.toInt).toArray
      for ( a <- 0 until flavors.length - 1 ) {
        for ( b <- a + 1 until flavors.length ) {
          if ( flavors(a) + flavors(b) == m ) {
            println(s"${a + 1} ${b + 1}")
          }
        }
      }
      //(0 until flavors.length - 1).flatMap(i => (i + 1 until flavors.length).filter(j => flavors(i) + flavors(j) == m))
    }
  }
}

File: Problems/hacker_rank/Search/20150122_Ice_Cream_Parlor/Solution.java
import java.util.Scanner;

public class Solution {
  public static void flavorIndex(final int[] flavors, final int money)  {
    for ( int i = 0; i < flavors.length - 1; ++i )  {
      for ( int j = i + 1; j < flavors.length; ++j )  {
        if ( flavors[i] + flavors[j] == money ) {
          System.out.println(String.format("%d %d", i + 1, j + 1));
          return;
        }
      }
    }
  }
  public static void main(final String[] args)  {
    Scanner sc = new Scanner(System.in);
    final int t = Integer.parseInt(sc.nextLine());
    for ( int i = 0; i < t; ++i ) {
      final int m = Integer.parseInt(sc.nextLine());
      final int n = Integer.parseInt(sc.nextLine());
      String[] inps = sc.nextLine().split(" ");
      final int[] flavors = new int[inps.length];
      for ( int j = 0; j < flavors.length; ++j )
        flavors[j] = Integer.parseInt(inps[j]);
      flavorIndex(flavors, m);
    }
  }
}
